code/UpdateOrder.js

```diff
@@ -60,13 +60,17 @@ function remove(order, removedItems) {
    if (!removedItems) {
      return order
    }
-   [].concat(removedItems.items).forEach(function(removedItems) {
-     var index = lib.findItemIndex(order, removedItems.item)
+   [].concat(removedItems.items).forEach(function(removedItem) {
+     var index = lib.findItemIndex(order, removedItem.item)
      if (index >= 0) {
        // deep copy of order.totalPrice, because it refers to the same object as order.items[index].shirt.price for some reason...
        order.totalPrice = Object.assign({}, order.totalPrice)
-       order.totalPrice.value -= order.items[index].shirt.price.value*order.items[index].quantity
-       order.items.splice(index, 1)
+       order.totalPrice.value -= order.items[index].shirt.price.value*removedItem.removedQuantity;
+       order.items[index].quantity -= removedItem.removedQuantity;
+       // delete the item if the quantity is zero
+       if (order.items[index].quantity === 0) {
```

Comment:
now that i look at this code, do you protect against people removing more shirts than exist?

Reply:
oh whoops. I meant to do a <= 0 check. It was in my head but I forgot write it out.

You can say remove 100 shirts and if you had 1, it would be -99 with a `=== 0` check. Ill fix it

Reply:
Also I have to protect against that for updating the total Price. There's a lot more to do here...

Reply:
I think it's easier to cap the number of shirts you can remove at the top, and the rest of the logic can kind of stay

Reply:
oh whoops didnt see this until after i finished a fix.
its alright this should work.

Reply:
Re approved